src/common/components/repository-row/index.js

```diff
@@ -212,7 +224,8 @@ export class RepositoryRowView extends Component {
       registerNameStatus
     } = this.props;
 
-    const showUnconfiguredDropdown = this.state.unconfiguredDropdownExpanded;
+    const showUnconfiguredDropdown = !snapIsConfigured(snap) && this.state.unconfiguredDropdownExpanded;
```

Comment:
This originally removed the unused `configured` variable and replaced it with a new function added for this feature. I added this because I found the dropdown remains open if a snapcraft.yml file is added.